### PR TITLE
Add note about OAuth2

### PR DESCRIPTION
**Pull request by VFDan**

### CODE REVIEW

**Review by Kenny2github — CHANGES_REQUESTED** [review]
The README is generated from the module docstring in the `__init__.py` file. Please modify that instead, then run `setup.py` with no arguments which will update the README.

**Comment by Kenny2github** [comment]
I'm not so sure it's actually recommended to turn on "require OAuth2 code grant". Not every app has the ability to have a public-facing redirect URI. You just have to make sure that your OAuth2 URL has `applications.commands` as one of the scopes.

File: README.rst [hunk 1]
```diff
@@ -47,6 +47,7 @@ Notes
   completely without the involvement of messages. However, channel and author
   information is still available.
 * All descriptions are **required**.
+* You must grant the bot `applications.commands` permissions in the OAuth2 section of the dashboard. It is recommended to also turn on "Requires OAuth2 Code Grant" in the Bot section of the developer dashboard.
```
Comment (Kenny2github) [inline README.rst]:
```suggestion
* You must grant the bot `applications.commands` permissions in the OAuth2 section of the developer dashboard.
```

**Review by Kenny2github — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (VFDan) [reply README.rst]: If I do this I don't have to build it with `setup.py`?

Comment (Kenny2github) [reply README.rst]: you don't

**Review by Kenny2github — APPROVED** [review]

**Pull request merged**